var/spack/repos/builtin/packages/parallel-netcdf/package.py

```diff
@@ -153,46 +154,46 @@ def configure_args(self):
     def cache_test_sources(self):
         """Copy the example source files after the package is installed to an
         install test subdirectory for use during `spack test run`."""
-        self.cache_extra_test_sources([self.examples_src_dir])
+        cache_extra_test_sources(self, [self.examples_src_dir])
 
-    def test(self):
+    def test_column_wise(self):
+        """Test column_wise executable"""
         test_dir = join_path(self.test_suite.current_test_cache_dir, self.examples_src_dir)
         # pnetcdf has many examples to serve as a suitable smoke check.
         # column_wise was chosen based on the E4S test suite. Other
         # examples should work as well.
         test_exe = "column_wise"
         options = [
-            "{0}.cpp".format(test_exe),
+            f"{test_exe}.cpp",
             "-o",
             test_exe,
             "-lpnetcdf",
-            "-L{0}".format(self.prefix.lib),
-            "-I{0}".format(self.prefix.include),
-        ]
-        reason = "test: compiling and linking pnetcdf example"
-        self.run_test(
-            self.spec["mpi"].mpicxx,
-            options,
-            [],
-            installed=False,
-            purpose=reason,
-            work_dir=test_dir,
-        )
-        mpiexe_list = [
-            self.spec["mpi"].prefix.bin.srun,
-            self.spec["mpi"].prefix.bin.mpirun,
-            self.spec["mpi"].prefix.bin.mpiexec,
+            f"-L{self.prefix.lib}",
+            f"-I{self.prefix.include}",
         ]
-        for mpiexe in mpiexe_list:
-            if os.path.isfile(mpiexe):
-                self.run_test(
-                    mpiexe,
-                    ["-n", "1", test_exe],
-                    [],
-                    installed=False,
-                    purpose="test: pnetcdf smoke test",
-                    skip_missing=True,
-                    work_dir=test_dir,
-                )
-                break
-        self.run_test("rm", ["-f", test_exe], work_dir=test_dir)
+
+        with working_dir(test_dir):
+
+            exe = which(self.spec["mpi"].prefix.bin.mpicxx)
+            exe(*options)
```

Comment:
Since the executable is actually known and as an aid for those running `spack pkg grep`, better to be more specific in the variable name:

```suggestion
            mpicxx = which(self.spec["mpi"].prefix.bin.mpicxx)
            mpicxx(*options)
```